src/main/java/com/artipie/asto/cache/CacheControl.java

```diff
@@ -61,16 +67,56 @@
         }
 
         @Override
-        public CompletionStage<Boolean> validate(final Key item) {
-            return this.origin.validate(item);
+        public CompletionStage<Boolean> validate(final Key item,
+            final Supplier<? extends CompletionStage<? extends Content>> supplier) {
+            return this.origin.validate(item, supplier);
         }
     }
 
     /**
      * Validate cached item: checks if cached value can be used or needs to be updated by fresh
      * value.
      * @param item Cached item
+     * @param content Content supplier
      * @return True if cached item can be used, false if needs to be updated
      */
-    CompletionStage<Boolean> validate(Key item);
+    CompletionStage<Boolean> validate(Key item,
+        Supplier<? extends CompletionStage<? extends Content>> content);
+
+    /**
+     * All cache controls should validate the cache.
+     * @since 0.25
+     */
+    final class All implements CacheControl {
+
+        /**
+         * Cache control items.
+         */
+        private final Collection<CacheControl> items;
+
+        /**
+         * All of items should validate the cache.
+         * @param items Cache controls
+         */
+        public All(final CacheControl... items) {
+            this(Arrays.asList(items));
+        }
+
+        /**
+         * All of items should validate the cache.
+         * @param items Cache controls
+         */
+        public All(final Collection<CacheControl> items) {
+            this.items = items;
+        }
+
+        @Override
+        public CompletionStage<Boolean> validate(final Key key,
+            final Supplier<? extends CompletionStage<? extends Content>> content) {
+            return Observable.fromIterable(this.items)
+                .flatMapSingle(item -> SingleInterop.fromFuture(item.validate(key, content)))
+                .all(item -> item)
+                .to(SingleInterop.get());
```

Comment:
@g4s8 it might be easier to compose `item.validate(...)` futures using `CompletableFuture.allOf()` without using reactive abstractions

Reply:
@olegmoz I'm not sure it's possible, since `ompletableFuture.allOf()` reutrns `CompletableFuture<Void>` but I need `CopletableFuture<Boolean>` here

Reply:
@g4s8 oh, that's right, missed that